composed-true-concidered-harmful.md

```diff
@@ -1,35 +1,583 @@
+
+Native HTML elements communicate up the DOM tree using DOM events. You might be used to seeing this with elements like `<input />` which publish events like `change` and `input` or with the `<button />` element, where it's common to rely on the `click` event that it publishes. It might not be immediately clear you are relying on these things, but if you've worked with `onclick` (native) or `onChange` (virtual DOM) properties, it is these DOM events that you are relying on under the hood. Knowing that these events are dispatch along the DOM tree, we can choose locations (either explicit or general) at which to listen for the via the `addEventListener(type, listener[, options/useCapture])` method that is present on any `HTMLElement` based DOM node.
+
+These events have two phases; the "capture" phase and the "bubble" phase. During the capture phase, the event travels from the top of the DOM down towards the dispatching element and can be listened for on each of the elements that it passes through in this phase by setting the third argument of `addEventListener()` to true, or by explicitly including `capture: true` in an `options` object passed as the third argument. For example the steps of the "capture" phase of a `click` event on the `<button>` in the following DOM structure:
+
+```html
+<body>
+    <header>
+        <nav>
+            <button>Click me!</button>
+        </nav>
+    </header>
+</body>
+```
+
+Would be as follows:
+
+1. `<body>`
+2. `<header>`
+3. `<nav>`
+4. `<button>`
+
+Then, being a `click` event has `bubbles: true` by default, the event would enter the "bubble" phase of the event and travel back up the DOM passing through the above DOM in the following order:
+
+1. `<button>`
+2. `<nav>`
+3. `<header>`
+4. `<body>`
+
+At any point in either phase that you are listening for this event, you will have access to the `preventDefault()`, `stopPropagation()`, and `stopImmediatePropagation()` methods that give you powerful the events that travel across your application. `preventDefault()` can most clearly be felt when listening to a `click` event on an `<a href="...">` tag. In this context, it will _prevent_ the anchor link from being activated and prevent the page from navigating. In a way, this is the event asking for permission to do an action, and we'll look at this more closely in conjunction with manually dispatched events. `stopPropagation()` prevents the event in question from continuing along the DOM tree and triggering listeners along that path, a sort of escape valve for the event when certain parameters are met. This can be taken one step further via `stopImmediatePropagation()` which also prevents the event from completing the current step of the phase it is in. This means that no later bound listeners on that same DOM element for the event in question will be called. So for the `<button>` element in the example above, when a `click` event is dispatched, you could imagine the following completely trivial listeners:
```

Comment:
'that give you powerful the events'?

Reply:
`give you powerful control over the events` 🙏

Reply:
^ added suggestion for convenience :)